src/inmanta/parser/plyInmantaParser.py

```diff
@@ -692,6 +692,33 @@ def p_boolean_expression_is_defined_short(p: YaccProduction) -> None:
     attach_lnr(p)
 
 
+def p_boolean_expression_is_defined_map_lookup_0(p: YaccProduction) -> None:
+    """boolean_expression : is_defined_expr IS DEFINED"""
+    p[0] = p[1]
+
+
+def p_is_defined_expr_is_defined_map_lookup_1(p: YaccProduction) -> None:
+    """is_defined_expr : var_ref '[' operand ']'
+    | attr_ref '[' operand ']'
+    | map_lookup '[' operand ']'"""
+
+    location = Location(file, p.lineno(2))
+    lexpos = p.lexpos(2)
+
+    def attach_lnr_to_statement(inp: ExpressionStatement) -> ExpressionStatement:
+        "set location"
+        inp.location = location
+        inp.lexpos = lexpos
+        return inp
+
+    key_in_dict = attach_lnr_to_statement(In(p[3], p[1]))
+    not_none = attach_lnr_to_statement(NotEqual(MapLookup(p[1], p[3]), attach_lnr_to_statement(Literal(NoneValue()))))
+    not_empty_list = attach_lnr_to_statement(NotEqual(MapLookup(p[1], p[3]), attach_lnr_to_statement(CreateList(list()))))
```

Comment:
Shouldn't have too much of an impact but creating two instances of `MapLookup` actually creates double work for the excution. I'd use a variable here. (Solved automatically in case you reuse the `map_lookup` production.